storage-node/src/services/sync/synchronizer.ts

```diff
@@ -45,17 +45,26 @@ export async function performSync(
   selectedOperatorUrl?: string
 ): Promise<void> {
   logger.info('Started syncing...')
+  logger.info('Sync - Fetching obligations...')
+  // This large query is also done by cleanup service.. sync the two services to use the same promise to get result
+  // one-time if queries are concurrent.
   const [model, files] = await Promise.all([getStorageObligationsFromRuntime(qnApi, buckets), getDataObjectIDs()])
```

Comment:
When thinking only about "new" objects to fetch, it makes sense to fetch all obligations on first run.
On subsequent calls we should find an approach to limit the size of the data set.